messaging/src/test/java/org/axonframework/eventhandling/replay/ReplayAwareMessageHandlerWrapperWithDisallowReplayTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2010-2019. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling.replay;
+
+import org.axonframework.eventhandling.AllowReplay;
+import org.axonframework.eventhandling.AnnotationEventHandlerAdapter;
+import org.axonframework.eventhandling.DisallowReplay;
+import org.axonframework.eventhandling.EventHandler;
+import org.axonframework.eventhandling.GenericTrackedEventMessage;
+import org.axonframework.eventhandling.GlobalSequenceTrackingToken;
+import org.axonframework.eventhandling.ReplayToken;
+import org.axonframework.eventhandling.TrackingToken;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.axonframework.eventhandling.GenericEventMessage.asEventMessage;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class ReplayAwareMessageHandlerWrapperWithDisallowReplayTest {
```

Comment:
JavaDoc on what's being tested here would be nice.

Reply:
Tests in that module don't seem to have javadoc

Reply:
We should definitely improve in that area. At least let's do it for newly added tests :)